django_etebase/routers/authentication.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+
+import nacl
+import nacl.encoding
+import nacl.hash
+import nacl.secret
+import nacl.signing
+from django.conf import settings
+from django.contrib.auth import get_user_model, user_logged_out
+from fastapi import APIRouter, Depends, status, Request, Response
+from pydantic import BaseModel
+
+from django_etebase.authentication import get_authenticated_user, AuthData, get_auth_data
+from django_etebase.fastapi_msgpack import MsgpackRoute, MsgpackResponse
+from django_etebase.views import msgpack_encode
+
+authentication_router = APIRouter(route_class=MsgpackRoute)
+User = get_user_model()
+
+
+class LoginChallengeResponse(BaseModel):
+    salt: bytes
+    challenge: bytes
+    version: int
+
+
+def get_encryption_key(salt):
+    key = nacl.hash.blake2b(settings.SECRET_KEY.encode(), encoder=nacl.encoding.RawEncoder)
+    return nacl.hash.blake2b(
+        b"",
+        key=key,
+        salt=salt[: nacl.hash.BLAKE2B_SALTBYTES],
+        person=b"etebase-auth",
+        encoder=nacl.encoding.RawEncoder,
+    )
+
+
+@authentication_router.post("/login_challenge/")
+def login_challenge(user: User = Depends(get_authenticated_user)):
+    enc_key = get_encryption_key(user.userinfo.salt)
+    box = nacl.secret.SecretBox(enc_key)
+    challenge_data = {
+        "timestamp": int(datetime.now().timestamp()),
+        "userId": user.id,
+    }
+    challenge = bytes(box.encrypt(msgpack_encode(challenge_data), encoder=nacl.encoding.RawEncoder))
+    return MsgpackResponse({"salt": user.userinfo.salt, "version": user.userinfo.version, "challenge": challenge})
```

Comment:
What happens if you return `Response` with this data? Doesn't it automatically get converted to `MsgpackResponse`? If so, it may be better/nicer due to the API explorer.

Reply:
It would be nicer , still not sure how to do that , if i return this dict it will fail to serialize  to json , with fastapi encoder method.

Reply:
OK, so let's ignore it for now. I don't really care about JSON anyway, it was just a nice to have.